02-flatten-array.js

```diff
@@ -7,15 +7,16 @@
 export function flattenArray(arr) {
     // TODO: ISSUE #4242 : Our customer reported that this method doesn't work in the case of deeply nested arrays
     // I added the test case to the spec file, but I'm not sure how to fix it. Can you help?
-    const result = [];
-    for (let i = 0; i < arr.length; i++) {
-        if (Array.isArray(arr[i])) {
-            for (let j = 0; j < arr[i].length; j++) {
-                result.push(arr[i][j]);
-            }
-        } else {
-            result.push(arr[i]);
-        }
-    }
+    // const result = [];
+    // for (let i = 0; i < arr.length; i++) {
+    //     if (Array.isArray(arr[i])) {
+    //         for (let j = 0; j < arr[i].length; j++) {
+    //             result.push(arr[i][j]);
+    //         }
+    //     } else {
+    //         result.push(arr[i]);
+    //     }
+    // }
+    const result = arr.flat(Infinity);
```

Comment:
The customer updated its specifications : The use of arr.flat() is prohibited because they use an old browser that doesn't support it. Could you please rebase your work ?

(But OK, 10 points to griffindor for telling me about this method 😂 )